src/lib/plugin_apis/fs.api

```diff
@@ -361,6 +361,62 @@ gchar* bd_fs_get_mountpoint (const gchar *device, GError **error);
  */
 gboolean bd_fs_resize (const gchar *device, guint64 new_size, GError **error);
 
+typedef enum {
+    BD_FS_NO_SUPPORT,
+    BD_FS_OFFLINE,
+    BD_FS_OFFLINEGROW,
+    BD_FS_ONLINE,
+    BD_FS_ONLINEGROW,
+    BD_FS_ONLINE_AND_OFFLINE,
+    BD_FS_ONLINEGROW_AND_OFFLINE,
+    BD_FS_ONLINEGROW_AND_OFFLINEGROW,
+} BDFsOnlineOffline;
```

Comment:
These need to have values assigned. And I think clever values can be chosen (powers of 2) that would allow the flags to be combined with bitwise OR. That way not so many would be needed.